src/Akeneo/Pim/Enrichment/Bundle/Command/IndexProductModelCommand.php

```diff
@@ -209,6 +218,76 @@ private function getExistingproductModelCodes(array $productModelCodes): array
         )->fetchFirstColumn();
     }
 
+    private function getDiffProductModelCodes(int $batchSize): iterable
+    {
+        $formerId = 0;
+        $sql = <<< SQL
+SELECT CONCAT('product_model_', id) AS _id, id, code, DATE_FORMAT(updated, '%Y-%m-%dT%TZ') AS updated
+FROM pim_catalog_product_model
+WHERE id > :formerId
+AND parent_id IS NULL
+ORDER BY id ASC
+LIMIT :limit
+SQL;
+        while (true) {
+            $rows = $this->connection->executeQuery(
+                $sql,
+                [
+                    'formerId' => $formerId,
+                    'limit' => $batchSize
+                ],
+                [
+                    'formerId' => \PDO::PARAM_INT,
+                    'limit' => \PDO::PARAM_INT
+                ]
+            )->fetchAllAssociative();
+
+            if (empty($rows)) {
+                return;
+            }
+
+            $formerId = (int)end($rows)['id'];
+            $existingMysqlIdentifiers = array_column($rows, '_id');
+            $existingMysqlUpdated = array_column($rows, 'updated');
+
+            $results = $this->productAndProductModelClient->search([
+               'query' => [
+                   'bool' => [
+                       'must' => [
+                           'ids' => [
+                               'values' => $existingMysqlIdentifiers
+                           ]
+                       ],
+                       'filter' => [
+                           'terms' => [
+                               'entity_updated' => $existingMysqlUpdated
+                           ]
+                       ]
+                   ]
+               ],
+                '_source' => false,
+                'size' => $batchSize
+            ]);
+
+            $esIdentifiers = array_map(function ($doc) {
+                return $doc['_id'];
+            }, $results['hits']['hits']);
+
+            $diff = array_reduce(
```

Comment:
There is already a function to do what you want:
```suggestion
            $diff = array_diff($existingMysqlIdentifiers, $esIdentifiers);
```

Note: array_reduce is quite complicated to read in php.